lib/Loader.php

```diff
@@ -130,16 +131,29 @@ protected function template_exists( $name ) {
 	 * @return \Twig_Loader_Filesystem
 	 */
 	public function get_loader() {
-		$open_basedir = ini_get('open_basedir');
-		$paths = array_merge($this->locations, array($open_basedir ? ABSPATH : '/'));
-		$paths = apply_filters('timber/loader/paths', $paths);
+		$open_basedir = ini_get( 'open_basedir' );
+		$paths        = array_merge_recursive(
+			$this->locations,
+			array( FilesystemLoader::MAIN_NAMESPACE => $open_basedir ? ABSPATH : '/' )
+		);
+		$paths        = apply_filters( 'timber/loader/paths', $paths );
 
 		$rootPath = '/';
 		if ( $open_basedir ) {
 			$rootPath = null;
 		}
-		$fs = new \Twig_Loader_Filesystem($paths, $rootPath);
-		$fs = apply_filters('timber/loader/loader', $fs);
+		$fs = new \Twig_Loader_Filesystem( array(), $rootPath );
+		foreach ( $paths as $namespace => $path_locations ) {
+			if ( is_array( $path_locations ) ) {
+				array_map( function ( $path ) use ( $fs, $namespace ) {
+					$fs->addPath( $path, $namespace );
+				}, $path_locations );
+			} else {
+				$fs->addPath( $path_locations, $namespace );
+			}
```

Comment:
I'm being overly defensive here as `$path_locations` will always be an array, as it affects code coverage I'll remove this if/else.

Reply:
Have removed the else, think the if is fine.